### DISCUSSION TITLE
GraphQL API Protected branch

### DISCUSSION THREAD

**Discussion by parithiban** [community/community#24640] [Apps, API and Webhooks]
<p>I am using a <a href="https://developer.github.com/v3/" rel="nofollow noopener">GITHUB API V3</a> in one of my project now we are migrating to <a href="https://developer.github.com/v4" rel="nofollow noopener">GraraphQL API V4</a>. I want to list all the branches of the repo and i want to check wether it is protected branch. In <a href="https://developer.github.com/v3/" rel="nofollow noopener">GITHUB API V3</a> it has a <a href="https://developer.github.com/v3/repos/branches/" rel="nofollow noopener">branches</a> api that will list all the branches along with it is protected key so it was each to check. Now in GraphQL i can get all the branches list but i can’t check wether the branch is protected or not.</p>
<pre><code>refs(first: 100, refPrefix: "refs/heads/") {
        	nodes {
        		name
        	}
        }
</code></pre>
<p>Someone could assist on this how to proceed?</p>
<p>If that could not be achieved by the <code>refs</code> then is there a way to list all the protected branches alone</p>
<p>Regards</p>
<p>Parithiban</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>You can use the following query to get the list of branch protection rules (and what branches they match) and the list of all branches for a repository:</p>
<pre><code>{
  repository(owner: "atom", name: "atom") {
    branchProtectionRules(first: 100) {
      nodes {
        pattern
        matchingRefs(first: 100) {
          nodes {
            name
          }
        }
      }
    }
    refs(first: 100, refPrefix: "refs/heads/") {
      nodes {
        name
      }
    }
  }
}
</code></pre>
<p>I hope that helps! Let us know if you have more questions.</p>

**Comment by maryamghani** [comment]
<p>Hello, </p>
<p>how can i use the same kind of pattern to protect my branch with wildcard rules like: release*</p>
<p>I am trying. to use API call for this</p>

**Comment by parithiban** [comment]
<p><a class="mention" href="/maloochpich">@maloochpich</a></p>
<p>See the below example to create a branch protection rule. You can change *pari* with release*</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8eb5d44c0691393d54878756505180d5eb5bf89a.png" data-download-href="/uploads/short-url/kmtmPymKpWnixMiFy4g2bGdrOhY.png?dl=1" title="Screen Shot 2020-03-10 at 6 48 43 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/7281166/181085783-a85769f9-facc-404e-af6a-447ae90bed95.png" alt="Screen Shot 2020-03-10 at 6 48 43 PM.png" data-base62-sha1="kmtmPymKpWnixMiFy4g2bGdrOhY" width="690" height="195" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/8eb5d44c0691393d54878756505180d5eb5bf89a_2_690x195.png, https://aws1.discourse-cdn.com/github/optimized/2X/8/8eb5d44c0691393d54878756505180d5eb5bf89a_2_1035x292.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/8/8eb5d44c0691393d54878756505180d5eb5bf89a_2_1380x390.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8eb5d44c0691393d54878756505180d5eb5bf89a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-03-10 at 6 48 43 PM.png</span><span class="informations">2852×810 205 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>You can check the output in the settings tab and clicking branches options in your repo.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/58786e3dba88f14fccc82e57b7cb4678d02a7827.png" data-download-href="/uploads/short-url/cCE2pgZX70akLamKl3snssaDEBF.png?dl=1" title="Screen Shot 2020-03-10 at 6.49.30 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/7281166/181085784-afb039cc-25a0-4934-931d-f33e4e27fe66.png" alt="Screen Shot 2020-03-10 at 6.49.30 PM.png" data-base62-sha1="cCE2pgZX70akLamKl3snssaDEBF" width="690" height="358" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/58786e3dba88f14fccc82e57b7cb4678d02a7827_2_690x358.png, https://aws1.discourse-cdn.com/github/optimized/2X/5/58786e3dba88f14fccc82e57b7cb4678d02a7827_2_1035x537.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/5/58786e3dba88f14fccc82e57b7cb4678d02a7827_2_1380x716.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/58786e3dba88f14fccc82e57b7cb4678d02a7827_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-03-10 at 6.49.30 PM.png</span><span class="informations">2752×1430 363 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by maryamghani** [comment]
<p><a class="mention" href="/parithiban">@parithiban</a>   thanks a lot that’s what I was looking for</p>

**Comment by makeapp007** [comment]
<p>Does github have api to see who can push to the protected branch? I am looking at pushAllowances  but couldn’t figure it out.</p>
<p>The fix is to use fragment:</p>
<p>something like </p>
<pre><code>pushAllowances(first: 50) {
          nodes {
            actor {
              ... on App {
                app_name: name
              }
</code></pre>

**Comment by parithiban** [comment]
<p><a class="mention" href="/makeapp007">@makeapp007</a> </p>
<p>If I get your point correctly. You need to get the list of users who can push to the protected branch</p>
<p>You can follow the below steps:</p>
<ol>
<li>Create a Branch protection rule in your repo. You need to set the branch protections to the user/team/app <a href="https://help.github.com/en/enterprise/2.13/user/articles/enabling-branch-restrictions" rel="nofollow noopener">Reference</a>
</li>
</ol>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8a3795c0f409f3e4c553dfa75d4c236c0a8d624b.png" data-download-href="/uploads/short-url/jIIYPFjjjiU2g3fLVrN2P9Nuq8P.png?dl=1" title="Screen Shot 2020-04-22 at 12.06.02 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/7281166/181085788-711f85b4-e610-460b-8cf7-57c096b6eb9e.png" alt="Screen Shot 2020-04-22 at 12.06.02 PM.png" data-base62-sha1="jIIYPFjjjiU2g3fLVrN2P9Nuq8P" width="690" height="315" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/8a3795c0f409f3e4c553dfa75d4c236c0a8d624b_2_690x315.png, https://aws1.discourse-cdn.com/github/optimized/2X/8/8a3795c0f409f3e4c553dfa75d4c236c0a8d624b_2_1035x472.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/8/8a3795c0f409f3e4c553dfa75d4c236c0a8d624b_2_1380x630.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8a3795c0f409f3e4c553dfa75d4c236c0a8d624b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-04-22 at 12.06.02 PM.png</span><span class="informations">1458×666 70.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<ol start="2">
<li>You can call the branch protection query for graphql to list out the user who has access to the protected branch.</li>
</ol>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/271817f7f48e3efb380db28d043022091293f1ef.png" data-download-href="/uploads/short-url/5zQeZ7gpKR10alB1YOyAyxpM7i7.png?dl=1" title="Screen Shot 2020-04-22 at 12 00 14 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/7281166/181085789-3734cc26-ce60-4978-8d51-3ebe0e84585f.png" alt="Screen Shot 2020-04-22 at 12 00 14 PM.png" data-base62-sha1="5zQeZ7gpKR10alB1YOyAyxpM7i7" width="690" height="338" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/271817f7f48e3efb380db28d043022091293f1ef_2_690x338.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/271817f7f48e3efb380db28d043022091293f1ef_2_1035x507.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/2/271817f7f48e3efb380db28d043022091293f1ef_2_1380x676.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/271817f7f48e3efb380db28d043022091293f1ef_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-04-22 at 12 00 14 PM.png</span><span class="informations">2874×1408 331 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Hope this answers your query.</p>

**Comment by maryamghani** [comment]
<p><a class="mention" href="/parithiban">@parithiban</a> </p>
<p>question regarding to <a href="https://developer.github.com/v4/input_object/deletebranchprotectionruleinput/" rel="nofollow noopener">https://developer.github.com/v4/input_object/deletebranchprotectionruleinput/</a></p>
<p>what is the <code>branchProtectionRuleId</code> is?</p>
<p>I tried to add the repository node_id did not work,</p>
<p>any suggestion?</p>
<pre><code>mutation {
    deleteBranchProtectionRule(input: {
        branchProtectionRuleId: "node_id"
    })
}
</code></pre>

**Comment by parithiban** [comment]
<p><a class="mention" href="/maloochpich">@maloochpich</a> </p>
<p>First, you need to get the branch protection id then you can pass that to the mutation input parameter</p>
<p>To get the branch protection id see below</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c70c0c39c94ae1c38e96681bad51ba61308bdb76.png" data-download-href="/uploads/short-url/soQPnYOgwkPh9rFlysWpUM85wWi.png?dl=1" title="Screen_Shot_2020_05_05_at_1_11_59_PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/7281166/181085792-3b1c548f-3fb3-4269-9519-28ceb66cd844.png" alt="Screen_Shot_2020_05_05_at_1_11_59_PM.png" data-base62-sha1="soQPnYOgwkPh9rFlysWpUM85wWi" width="690" height="360" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/c70c0c39c94ae1c38e96681bad51ba61308bdb76_2_690x360.png, https://aws1.discourse-cdn.com/github/optimized/2X/c/c70c0c39c94ae1c38e96681bad51ba61308bdb76_2_1035x540.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/c/c70c0c39c94ae1c38e96681bad51ba61308bdb76_2_1380x720.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c70c0c39c94ae1c38e96681bad51ba61308bdb76_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen_Shot_2020_05_05_at_1_11_59_PM.png</span><span class="informations">2878×1504 346 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Then you need to pass that branch protection id.</p>
<pre><code>mutation {
	deleteBranchProtectionRule(input: {
		branchProtectionRuleId: "YOUR_BRANCH_PROTECTION_ID"
	})
}
</code></pre>
<p>By applying the above query your branch protection will be deleted.</p>

**Comment by maryamghani** [comment]
<p>how can we add user to the branch protection like this? <a class="mention" href="/parithiban">@parithiban</a><br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/434ccccc9ee191e5cb43b38b276a1cd3da44a8f6.png" data-download-href="/uploads/short-url/9Bmx5KQv5pjIfspQnFBsS5JXeJ0.png?dl=1" title="Screen Shot 2020-09-30 at 9.28.34 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/2685271/181085793-a5b2619f-ead0-4694-b504-ba763cf1588d.png" alt="Screen Shot 2020-09-30 at 9.28.34 PM" data-base62-sha1="9Bmx5KQv5pjIfspQnFBsS5JXeJ0" width="690" height="63" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/434ccccc9ee191e5cb43b38b276a1cd3da44a8f6_2_690x63.png, https://aws1.discourse-cdn.com/github/optimized/2X/4/434ccccc9ee191e5cb43b38b276a1cd3da44a8f6_2_1035x94.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/4/434ccccc9ee191e5cb43b38b276a1cd3da44a8f6_2_1380x126.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/434ccccc9ee191e5cb43b38b276a1cd3da44a8f6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-09-30 at 9.28.34 PM</span><span class="informations">1446×134 23.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by maryamghani** [comment]
<p>this is not working for me</p>

**Comment by parithiban** [comment]
<p><a class="mention" href="/maloochpich">@maloochpich</a></p>
<p>You can try with the following query</p>
<pre><code class="lang-auto">mutation{
  updateBranchProtectionRule(input:
    {
      branchProtectionRuleId:"YOUR_BRANCH_PROTECTION_ID",
      restrictsPushes:true,
      pushActorIds:["USER_ID","TEAM_ID"]
    }){    
  }
}
</code></pre>

**Reply by grossmane** [reply]
Hi, @parithiban
I try to add pushActorIds  or any other things like bypassPullRequestActorIds with graphQl, but I didn't have any success.
I found not much informations or posts for GitHub GraphQL, so I found this thread here.
maby you can help me:
here is my code:
```
fragment branchProtection on BranchProtectionRule {
  allowsDeletions
  allowsForcePushes
  creator {
    login
  }
  id
  blocksCreations
  isAdminEnforced
  requiresLinearHistory
  requiresConversationResolution
  requiredStatusCheckContexts
  requiredApprovingReviewCount
  requiresApprovingReviews
  requiresCodeOwnerReviews
  requiresStatusChecks
  restrictsPushes
  restrictsReviewDismissals
  dismissesStaleReviews
  pattern
}
mutation addBranchProtection(
  $repositoryId:                    ID!,
  $branchPattern:                   String!,
  $allowsDeletions:                 Boolean = false,
  $allowsForcePushes:               Boolean = false,
  $blocksCreations:                 Boolean = false,
  $dismissesStaleReviews:           Boolean = false,
  $isAdminEnforced:                 Boolean = true,
  $requiresLinearHistory:           Boolean = true,
  $requiresConversationResolution:  Boolean = false,
  $requiresApprovingReviews:        Boolean = false,
  $requiredApprovingReviewCount:    Int = 0,
  $requiresCodeOwnerReviews:        Boolean = false,
  $requiredStatusChecks:            [String!],
  $requiresStatusChecks:            Boolean = true,
  $restrictsReviewDismissals:       Boolean = false,
  $restrictsPushes:                 Boolean = false
) {
  createBranchProtectionRule(input: {
    repositoryId:                   $repositoryId
    pattern:                        $branchPattern
    allowsDeletions:                $allowsDeletions
    allowsForcePushes:              $allowsForcePushes
    blocksCreations:                $blocksCreations
    dismissesStaleReviews:          $dismissesStaleReviews
    isAdminEnforced:                $isAdminEnforced
    requiresLinearHistory:          $requiresLinearHistory
    requiresConversationResolution: $requiresConversationResolution
    requiresApprovingReviews:       $requiresApprovingReviews
    requiredApprovingReviewCount:   $requiredApprovingReviewCount
    pushActorIds:                   ["MYUSERID"]
    requiresCodeOwnerReviews:       $requiresCodeOwnerReviews
    requiredStatusCheckContexts:    $requiredStatusChecks
    requiresStatusChecks:           $requiresStatusChecks
    restrictsReviewDismissals:      $restrictsReviewDismissals
    restrictsPushes:                $restrictsPushes
  }) {
    branchProtectionRule {
      ...branchProtection
    }
  }
}
```
and the error is:
```
{
  "data": {
    "createBranchProtectionRule": null
  },
  "errors": [
    {
      "type": "NOT_FOUND",
      "path": [
        "createBranchProtectionRule"
      ],
      "locations": [
        {
          "line": 60,
          "column": 3
        }
      ],
      "message": "Could not resolve to a node with the global id of 'MYUSERID'"
    }
  ]
}
```

note: i replace my real userID with "MYUSERID" in code snippets.

I hope you can help me.
many thanks,
Jens

**Reply by grossmane** [reply]
I found self the solution with the following code:
```
fragment branchProtection on BranchProtectionRule {
  pushAllowances(first:10) {
    nodes {
      id
    }
  }
  allowsDeletions
  allowsForcePushes
  creator {
    login
  }
  id
  blocksCreations
  isAdminEnforced
  requiresLinearHistory
  requiresConversationResolution
  requiredStatusCheckContexts
  requiredApprovingReviewCount
  bypassPullRequestAllowances(first:10) {
    nodes {
      id
    }
  }
  requiresApprovingReviews
  requiresCodeOwnerReviews
  requiresStatusChecks
  restrictsPushes
  restrictsReviewDismissals
  dismissesStaleReviews
  pattern
}

query getUser($login: String!) {
  user(login: $login) {
    id
  }
}

mutation addBranchProtection(
  $repositoryId:                    ID!,
  $branchPattern:                   String!,
  $pushActorIds:                    [ID!] = [],
  $allowsDeletions:                 Boolean = false,
  $allowsForcePushes:               Boolean = false,
  $blocksCreations:                 Boolean = false,
  $dismissesStaleReviews:           Boolean = false,
  $isAdminEnforced:                 Boolean = true,
  $requiresLinearHistory:           Boolean = true,
  $requiresConversationResolution:  Boolean = false,
  $requiresApprovingReviews:        Boolean = false,
  $requiredApprovingReviewCount:    Int = 0,
  $bypassPullRequestActorIds:       [ID!] = [],
  $requiresCodeOwnerReviews:        Boolean = false,
  $requiredStatusChecks:            [String!],
  $requiresStatusChecks:            Boolean = true,
  $restrictsReviewDismissals:       Boolean = false,
  $restrictsPushes:                 Boolean = false
) {
  createBranchProtectionRule(input: {
    repositoryId:                   $repositoryId
    pattern:                        $branchPattern
    pushActorIds:                   $pushActorIds
    allowsDeletions:                $allowsDeletions
    allowsForcePushes:              $allowsForcePushes
    blocksCreations:                $blocksCreations
    dismissesStaleReviews:          $dismissesStaleReviews
    isAdminEnforced:                $isAdminEnforced
    requiresLinearHistory:          $requiresLinearHistory
    requiresConversationResolution: $requiresConversationResolution
    requiresApprovingReviews:       $requiresApprovingReviews
    requiredApprovingReviewCount:   $requiredApprovingReviewCount
    bypassPullRequestActorIds:      $bypassPullRequestActorIds
    requiresCodeOwnerReviews:       $requiresCodeOwnerReviews
    requiredStatusCheckContexts:    $requiredStatusChecks
    requiresStatusChecks:           $requiresStatusChecks
    restrictsReviewDismissals:      $restrictsReviewDismissals
    restrictsPushes:                $restrictsPushes
  }) {
    branchProtectionRule {
      ...branchProtection
    }
  }
}
```

as first request, I get the node ID of the user getUser() and then I request the addBranchProtection()

**Comment by maryamghani** [comment]
<aside class="quote no-group" data-username="parithiban" data-post="12" data-topic="14380">
<div class="title">
<div class="quote-controls"></div>
 parithiban:</div>
<blockquote>
<pre><code class="lang-auto">mutation{
  updateBranchProtectionRule(input:
    {
      branchProtectionRuleId:"YOUR_BRANCH_PROTECTION_ID",
      restrictsPushes:true,
      pushActorIds:["USER_ID","TEAM_ID"]
    }){    
  }
}
</code></pre>
</blockquote>
</aside>
<p><a class="mention" href="/parithiban">@parithiban</a>  this does not work</p>
<pre><code class="lang-auto">mutation{
  updateBranchProtectionRule(input:
    {
      branchProtectionRuleId:"",
      restrictsPushes:true,
      pushActorIds:[""]
    }){    
        branchProtectionRule{
            branchProtectionRuleId
            restrictsPushes
        }
  }
}
</code></pre>
<pre><code class="lang-auto">{
    "errors": [
        {
            "path": [
                "mutation",
                "updateBranchProtectionRule",
                "branchProtectionRule",
                "branchProtectionRuleId"
            ],
            "extensions": {
                "code": "undefinedField",
                "typeName": "BranchProtectionRule",
                "fieldName": "branchProtectionRuleId"
            },
            "locations": [
                {
                    "line": 9,
                    "column": 13
                }
            ],
            "message": "Field 'branchProtectionRuleId' doesn't exist on type 'BranchProtectionRule'"
        }
    ]
}
</code></pre>

**Comment by parithiban** [comment]
<p><a class="mention" href="/maloochpich">@maloochpich</a></p>
<p>You need to slightly modify your query</p>
<pre><code class="lang-auto">mutation{
 updateBranchProtectionRule(input:
   {
     branchProtectionRuleId:"",
     restrictsPushes:true,
     pushActorIds:[""]
   }){    
       branchProtectionRule{
           id
           restrictsPushes
       }
 }
}
</code></pre>
<p><a href="https://developer.github.com/v4/object/branchprotectionrule/" rel="noopener nofollow ugc">branchProtectionRule</a> node has id paramaeter and not branchProtectionRuleId</p>

**Comment by maryamghani** [comment]
<p>This worked <a class="mention" href="/parithiban">@parithiban</a> Thanks</p>

**Comment by maryamghani** [comment]
<p><a class="mention" href="/parithiban">@parithiban</a> by running this i get this error:</p>
<pre><code class="lang-auto">"message": "Field must have selections (field 'deleteBranchProtectionRule' returns DeleteBranchProtectionRulePayload but has no selections. Did you mean 'deleteBranchProtectionRule { ... }'?)"
</code></pre>

**Comment by parithiban** [comment]
<p><a class="mention" href="/maloochpich">@maloochpich</a></p>
<p>What is the mutation query that you are trying to execute?</p>

**Comment by maryamghani** [comment]
<p><a class="mention" href="/parithiban">@parithiban</a></p>
<pre><code class="lang-auto">mutation {
	deleteBranchProtectionRule(input: {
		branchProtectionRuleId: ""
	})
}
</code></pre>
<p>this is what i run</p>

**Comment by parithiban** [comment]
<p><a class="mention" href="/maloochpich">@maloochpich</a></p>
<p>As the error suggests it should have a selection of subfields. Can you try with the below query?</p>
<pre><code class="lang-auto">mutation {
   deleteBranchProtectionRule(input: {
   	branchProtectionRuleId: "XXXX-XXXX-XXXX"
 }){
   clientMutationId
 }
}
</code></pre>

**Comment by maryamghani** [comment]
<p>that is working thanks a lot for your help and being so helpful as always</p>

**Comment by navachaitanyak** [comment]
can someone help to get the branch protection rule id for one of the rules like, I have branch protection rule 'test*' and I want to get the branch protection rule id for this one, can someone help me, please
